flusher.go

```diff
@@ -584,7 +584,7 @@ func (s *Server) flushEventsChecks() {
 // this function - probably a static string for each callsite
 // you can disable compression with compress=false for endpoints that don't
 // support it
-func (s *Server) postHelper(ctx context.Context, endpoint string, bodyObject interface{}, action string, compress bool) error {
+func postHelper(ctx context.Context, s VeneurServer, endpoint string, bodyObject interface{}, action string, compress bool) error {
```

Comment:
The only reason we're passing VeneurServer in is to get access to the statsd client, yes? Should we pass that in instead?

Or, even better - store it in the context?